src/ring_theory/adjoin/basic.lean

```diff
@@ -263,6 +309,15 @@ theorem adjoin_eq_ring_closure (s : set A) :
   (adjoin R s).to_subring = subring.closure (set.range (algebra_map R A) ∪ s) :=
 subring.ext $ λ x, mem_adjoin_iff
 
+variables (R)
+
+/-- If all elements of `s : set A` commute pairwise, then `adjoin R s` is a commutative
+ring.  -/
+def adjoin_comm_ring_of_comm {s : set A} (hcomm : ∀ (a ∈ s) (b ∈ s), a * b = b * a) :
```

Comment:
Can you generalize this for semirings? The proof is literally the same I think.

Also, d we have something similar for subgroups/submonoids?

Reply:
OK, I think I did all of them now!